tests/test_parsers/test_pdf_plumber_parser.py

```diff
@@ -24,4 +25,27 @@ def test_parse(self):
         # roughly the right content
         for keyword in ['Field', 'Task', 'SOTA', 'Base', 'Frozen', 'Finetune', 'NER']:
             assert keyword in doc.symbols[:100]
-        
\ No newline at end of file
+
+    def test_split_punctuation(self):
+        no_split_parser = PDFPlumberParser(split_at_punctuation=False)
+        no_split_doc = no_split_parser.parse(input_pdf_path="../fixtures/2107.07170.pdf")
+        no_split_tokens_with_numbers = [token.text for token in no_split_doc.tokens if re.search(r'[0-9]', token.text)]
+        assert '[1-5]' in no_split_tokens_with_numbers
+        assert 'GPT-3[10]' in no_split_tokens_with_numbers
+
+        split_parser = PDFPlumberParser(split_at_punctuation=True)
+        split_doc = split_parser.parse(input_pdf_path="../fixtures/2107.07170.pdf")
+        split_tokens_with_numbers = [token.text for token in split_doc.tokens if re.search(r'[0-9]', token.text)]
+        assert '[1-5]' not in split_tokens_with_numbers
+        assert '1-5' in split_tokens_with_numbers
+        assert 'GPT-3[10]' not in split_tokens_with_numbers
+        assert 'GPT-3' in split_tokens_with_numbers
```

Comment:
so, does it split like: "GPT-3", "[", "10", "]" ?

Reply:
yep!